src/util.cpp

```diff
@@ -758,15 +789,33 @@ void ArgsManager::ReadConfigStream(std::istream& stream)
     {
         std::string strKey = std::string("-") + it->string_key;
         std::string strValue = it->value[0];
+
         if (InterpretNegatedOption(strKey, strValue)) {
             m_config_args[strKey].clear();
         } else {
             m_config_args[strKey].push_back(strValue);
         }
+
+        // Check that the arg is known
+        bool found = false;
+        for (auto& arg : m_available_args) {
+            std::string arg_name = arg.first.second;
+            size_t eq_index = arg_name.find('=');
+            if (eq_index != std::string::npos) {
+                arg_name.erase(eq_index);
+            }
+            if (arg_name == ArgsManagerHelper::GetArgWithoutNet(strKey)) {
+                found = true;
+                break;
+            }
+        }
+        if (!found && !ignore_invalid_keys) {
+            throw std::runtime_error(strprintf("Invalid configuration value %s", it->string_key.c_str()));
```

Comment:
Wouldn't it be more straightforward to `return false;` instead of throwing an exception here? If you need to pass a string you could pass in a non-const reference to an `std::string`.

Reply:
Also, why is it required in the first place to ignore invalid keys for the cli?

Reply:
Using an exception for control flow is just another instance of "goto", imo. Even in python (where using exceptions for control flow is considered pythonic) we had serious issues that took several years to fix up completely.
If you prefer to pass a result object with a boolean indicating success and an optional error string, I think that is fine. But using an std::runtime_error as the class for this object seems confusing and is not type safe.